lib/OnyxUtils.ts

```diff
@@ -1041,6 +1050,30 @@ function initializeWithDefaultKeyStates(): Promise<void> {
     });
 }
 
+function isValidMergeCollection<TKey extends CollectionKeyBase, TMap>(collectionKey: TKey, collection: Collection<TKey, TMap, NullishDeep<KeyValueMapping[TKey]>>) {
```

Comment:
Also how about this declaration?
```suggestion
function isValidMergeCollection<TKey extends CollectionKeyBase>(collectionKey: TKey, collection: Record<string, unknown>): boolean {
```

We can use `Record<string, unknown>` here because this function verifies that `collection` is a valid argument for `enqueueMergeOperation` function